74-patch/arch/x86/coco/tdx/tdx.c
--- linux-6.12.74-orig/arch/x86/coco/tdx/tdx.c	2026-02-19 15:29:56.000000000 +0000
+++ linux-6.12.74-patch/arch/x86/coco/tdx/tdx.c	2026-02-27 15:08:37.102044599 +0000
@@ -1091,5 +1091,12 @@
 	 */
 	x86_cpuinit.parallel_bringup = false;
 
+	/*
+	 * Override init functions that scan the ROM region in TDX guests, as this
+	 * memory is not accepted and would thus cause a crash.
+	 */
+	pr_info("Skip probe_roms for TDX guest");
+	x86_init.resources.probe_roms = x86_init_noop;
+
 	pr_info("Guest detected\n");
 }
