85-patch/arch/x86/coco/tdx/tdx.c
--- linux-6.12.85-orig/arch/x86/coco/tdx/tdx.c	2026-05-01 08:46:41.093656490 +0000
+++ linux-6.12.85-patch/arch/x86/coco/tdx/tdx.c	2026-05-01 09:35:39.959022321 +0000
@@ -1091,5 +1091,12 @@
 	 */
 	x86_cpuinit.parallel_bringup = false;
 
+	/*
+	 * Override init functions that scan the ROM region in TDX guests, as this
+	 * memory is not accepted and would thus cause a crash.
+	 */
+	pr_info("Skip probe_roms for TDX guest");
+	x86_init.resources.probe_roms = x86_init_noop;
+
 	pr_info("Guest detected\n");
 }
