diff --git a/src/MODULE.bazel b/src/MODULE.bazel
index 1d181bee4..304801f55 100644
--- MODULE.bazel
+++ MODULE.bazel
@@ -30,3 +30,4 @@ bazel_dep(name = "googletest", version = "1.14.0.bcr.1")
 
 bazel_dep(name = "platforms", version = "0.0.9")
 bazel_dep(name = "rules_cc", version = "0.0.9")
+bazel_dep(name = "zlib", version = "1.3.1")
diff --git a/src/rust/bssl-crypto/BUILD b/src/rust/bssl-crypto/BUILD
new file mode 100644
index 000000000..61ae2afad
--- /dev/null
+++ b/src/rust/bssl-crypto/BUILD
@@ -0,0 +1,15 @@
+load("@rules_rust//rust:defs.bzl", "rust_library", "rust_test")
+
+rust_library(
+    name = "bssl-crypto",
+    srcs = glob(include = ["src/**/*.rs"]),
+    deps = [
+        "//src/rust/bssl-sys:bssl-sys",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+rust_test(
+    name = "bssl-crypto_test",
+    crate = ":bssl-crypto",
+)
diff --git a/src/rust/bssl-sys/BUILD b/src/rust/bssl-sys/BUILD
new file mode 100644
index 000000000..d6c5fed8b
--- /dev/null
+++ b/src/rust/bssl-sys/BUILD
@@ -0,0 +1,32 @@
+load("@rules_rust//bindgen:defs.bzl", "rust_bindgen_library")
+
+rust_bindgen_library(
+    name = "bssl-sys",
+    bindgen_flags = [
+        "--no-derive-default",
+        "--enable-function-attribute-detection",
+        "--use-core",
+        "--default-macro-constant-type=signed",
+        "--rustified-enum=point_conversion_form_t",
+        "--allowlist-file=.*[[:punct:]]openssl[[:punct:]].*\\.h",
+    ],
+    wrap_static_fns = True,
+    cc_lib = ":wrapper",
+    header = "wrapper.h",
+    visibility = [
+        "//src/rust:__subpackages__",
+    ],
+)
+
+cc_library(
+    name = "wrapper",
+    srcs = [
+        "rust_wrapper.c",
+        "rust_wrapper.h",
+    ],
+    hdrs = ["wrapper.h"],
+    deps = [
+        "//:crypto",
+        "//:ssl",
+    ],
+)

