diff --git tensorflow/lite/micro/debug_log.cc tensorflow/lite/micro/debug_log.cc
index 46ca253..d34dc21 100644
--- tensorflow/lite/micro/debug_log.cc
+++ tensorflow/lite/micro/debug_log.cc
@@ -45,6 +45,5 @@ extern "C" void DebugLog(const char* s) {
   // Reusing TF_LITE_STRIP_ERROR_STRINGS to disable DebugLog completely to get
   // maximum reduction in binary size. This is because we have DebugLog calls
   // via TF_LITE_CHECK that are not stubbed out by TF_LITE_REPORT_ERROR.
-  fprintf(stderr, "%s", s);
 #endif
 }

