From 53bb2207cc8975237f0a4b3442044c6b19b31a2a Mon Sep 17 00:00:00 2001
From: Tiziano Santoro <tiziano88@gmail.com>
Date: Wed, 31 Oct 2018 10:45:29 +0000
Subject: [PATCH 02/10] Make bazel rule public

---
 BUILD | 1 +
 1 file changed, 1 insertion(+)

diff --git BUILD BUILD
index 894a5997..13faa305 100644
--- BUILD
+++ BUILD
@@ -23,4 +23,5 @@ cc_library(
         "src/prebuilt/wasm2c.include.c",
         "src/range.h",
     ],
+    visibility = ["//visibility:public"],
 )
-- 
2.19.0.605.g01d371f741-goog

