From a21af7a5cf23d4a1b744a4fb80485d881978275e Mon Sep 17 00:00:00 2001
From: Tiziano Santoro <tiziano88@gmail.com>
Date: Wed, 31 Oct 2018 12:20:00 +0000
Subject: [PATCH 06/10] Try using cxx11 version of to_string

---
 src/binary-reader-ir.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/binary-reader-ir.cc src/binary-reader-ir.cc
index 62b03a83..0ce69a5b 100644
--- src/binary-reader-ir.cc
+++ src/binary-reader-ir.cc
@@ -1060,7 +1060,7 @@ Result BinaryReaderIR::OnFunctionName(Index index, string_view name) {
   int counter = 1;
   std::string orig_name = dollar_name;
   while (module_->func_bindings.count(dollar_name) != 0) {
-    dollar_name = orig_name + "." + std::to_string(static_cast<int>(counter++));
+    dollar_name = orig_name + "." + std::__cxx11::to_string(counter++);
   }
   func->name = dollar_name;
   module_->func_bindings.emplace(dollar_name, Binding(index));
-- 
2.19.0.605.g01d371f741-goog

