From 2d31cda394fc67c7969a9bd44066cb8eafa82e23 Mon Sep 17 00:00:00 2001
From: Tiziano Santoro <tiziano88@gmail.com>
Date: Wed, 31 Oct 2018 13:29:11 +0000
Subject: [PATCH 07/10] Include opcode-code-table.c

---

diff --git src/opcode-code-table.c src/opcode-code-table.c
index f830e34b..92126a18 100644
--- src/opcode-code-table.c
+++ src/opcode-code-table.c
@@ -16,7 +16,7 @@

 #include "src/opcode-code-table.h"

-#include "config.h"
+/*#include "config.h"*/

 #include <stdint.h>

@@ -29,7 +29,7 @@ typedef enum WabtOpcodeEnum {
   Invalid,
 } WabtOpcodeEnum;

-WABT_STATIC_ASSERT(Invalid <= WABT_OPCODE_CODE_TABLE_SIZE);
+/*WABT_STATIC_ASSERT(Invalid <= WABT_OPCODE_CODE_TABLE_SIZE);*/

 /* The array index calculated below must match the one in Opcode::FromCode. */
 uint32_t WabtOpcodeCodeTable[WABT_OPCODE_CODE_TABLE_SIZE] = {
--
2.19.0.605.g01d371f741-goog

