From fa9f27fafb4d21f1c3385392bea8260a3c79c4d9 Mon Sep 17 00:00:00 2001
From: David Drysdale <drysdale@google.com>
Date: Tue, 16 Jul 2019 07:34:09 +0100
Subject: [PATCH 08/10] Add public BUILD targets for wat2wasm,wasm2wat

---
 BUILD | 38 ++++++++++++++++++++++++++++----------
 1 file changed, 28 insertions(+), 10 deletions(-)

diff --git BUILD BUILD
index 71534062..1115fcdc 100644
--- BUILD
+++ BUILD
@@ -1,15 +1,21 @@
+package(
+    default_visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "wabt",
-    srcs = glob([
-        "src/*.cc",
-        "src/*.c",
-        "src/interp/*.cc",
-        "src/prebuilt/*.cc",
-    ],
-    exclude=[
-        "src/test-*",
-        "src/wast-lexer.cc",
-    ]),
+    srcs = glob(
+        [
+            "src/*.cc",
+            "src/*.c",
+            "src/interp/*.cc",
+            "src/prebuilt/*.cc",
+        ],
+        exclude = [
+            "src/test-*",
+            "src/wast-lexer.cc",
+        ],
+    ),
     hdrs = glob([
         "config.h",
         "src/*.h",
@@ -26,3 +32,15 @@ cc_library(
     ],
     visibility = ["//visibility:public"],
 )
+
+cc_binary(
+    name = "wat2wasm",
+    srcs = ["src/tools/wat2wasm.cc"],
+    deps = [":wabt"],
+)
+
+cc_binary(
+    name = "wasm2wat",
+    srcs = ["src/tools/wasm2wat.cc"],
+    deps = [":wabt"],
+)
-- 
2.19.0.605.g01d371f741-goog

