From 910d762920ac7575d7e3e04791a9d9c0ae07f8cd Mon Sep 17 00:00:00 2001
From: David Drysdale <drysdale@google.com>
Date: Tue, 16 Jul 2019 07:49:17 +0100
Subject: [PATCH 09/10] Only warn on NDEBUG build

---
 config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git config.h config.h
index df9a34e2..d9c68558 100644
--- config.h
+++ config.h
@@ -3,7 +3,7 @@
 
 // BEGIN GOOGLE3 MODIFICATIONS
 #ifdef NDEBUG
-#error This library uses assert()
+#warning This library uses assert()
 #endif
 // END GOOGLE3 MODIFICATIONS
 
-- 
2.19.0.605.g01d371f741-goog

