From ba7d28fc6aeee6b0733f0559a79f2d3a38b18d16 Mon Sep 17 00:00:00 2001
From: David Drysdale <dmd@lurklurk.org>
Date: Thu, 17 Dec 2020 17:37:59 +0000
Subject: [PATCH] Patch to remove mention of bundled protoc

---
 prost-build/build.rs | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/prost-build/build.rs b/prost-build/build.rs
index d5586516e1a4..e3e4ac9b496b 100644
--- a/prost-build/build.rs
+++ b/prost-build/build.rs
@@ -37,16 +37,7 @@ fn env_protoc() -> Option<PathBuf> {
 
 /// Returns the path to the bundled `protoc`, if it is available for the host platform.
 fn bundled_protoc() -> Option<PathBuf> {
-    let protoc_bin_name = match (env::consts::OS, env::consts::ARCH) {
-        ("linux", "x86") => "protoc-linux-x86_32",
-        ("linux", "x86_64") => "protoc-linux-x86_64",
-        ("linux", "aarch64") => "protoc-linux-aarch_64",
-        ("macos", "x86_64") => "protoc-osx-x86_64",
-        ("windows", _) => "protoc-win32.exe",
-        _ => return None,
-    };
-
-    Some(bundle_path().join(protoc_bin_name))
+    None
 }
 
 /// Returns the path to the `protoc` included on the `PATH`, if it exists.
-- 
2.26.2

