From a22d6c33caaa2cd9ba2c6a3c3401ebb02370b71d Mon Sep 17 00:00:00 2001
From: Conrad Grobler <grobler@google.com>
Date: Fri, 8 Oct 2021 12:49:53 +0000
Subject: [PATCH] Patch prost to allow Cargo.lock files to be committed

---
 third_party/prost/.gitignore | 1 -
 1 file changed, 1 deletion(-)

diff --git a/third_party/prost/.gitignore b/third_party/prost/.gitignore
index a9d37c56..eb5a316c 100644
--- a/third_party/prost/.gitignore
+++ b/third_party/prost/.gitignore
@@ -1,2 +1 @@
 target
-Cargo.lock
-- 
2.33.0.685.g46640cef36-goog
